src/lightning.rs

```diff
@@ -124,14 +126,14 @@ impl GetInfo<InvoiceInfo> for Invoice {
 							Fallback::SegWitProgram {
 								version: v,
 								program: p,
-							} => Payload::WitnessProgram {
-								version: WitnessVersion::try_from(v.to_u8())
+							} =>
+							Payload::WitnessProgram(WitnessProgram::new(
+								WitnessVersion::try_from(v.to_u8())
 									.expect("invalid segwit version in invoice"),
-								program: p.to_vec(),
-							},
+								p.to_vec(),
+							).expect("invalid segwit program")),
```

Comment:
Hmm, unfortunate that we don't have `.need()` available here outside of the binary. I think it's fine, though.